documentation/src/main/asciidoc/topics/con_spring_cache_annotations.adoc

```diff
@@ -0,0 +1,35 @@
+[id='spring-cache-annotations_{context}']
+= Spring Cache annotations
+Annotation application code with the `@Cacheable` and `@CacheEvict` annotations.
+
+.`@Cacheable`
+
+The `@Cacheable` annotation adds returned values to a defined cache.
+
+For instance, you have a data access object (DAO) for books. You want book instances to be cached after they have been loaded from the underlying database using `BookDao#findBook(Integer bookId)`.
+
+Annotate the `findBook(Integer bookId)` method with `@Cacheable` as follows:
+
+[source,java,options=nowrap]
+----
+include::code_examples/SpringCachableExample.java[]
+----
+
+Any `Book` instances returned from `findBook(Integer bookId)` are stored in a cache named `books`, using `bookId` as the key.
+
+Note that "#bookId" is an expression in the link:http://static.springsource.org/spring/docs/current/spring-framework-reference/html/expressions.html[Spring Expression Language] that evaluates the `bookId` argument.
+
+[IMPORTANT]
+====
+If your application needs to reference entries in the cache directly, you should include the `key` attribute. Without this attribute, Spring generates a hash from the supplied method arguments to use as the cache key.
```

Comment:
The default key generator does not use a hash of the arguments, it aggregates the arguments themselves in a SimpleKey:
https://docs.spring.io/spring-framework/docs/current/reference/html/integration.html#cache-annotations-cacheable-default-key

OTOH the SimpleKey will delegate equals/hashCode to the parameters, so the key attribute is needed if one of the parameters doesn't implement them (or cannot be serialized, or is just too big to be stored in the cache).

Reply:
@danberindei I updated this note to drop the detail about Spring behaviour and included that link in your comment. Could you please take another look? Thanks for the review!